031-custom_authentication.md

```diff
@@ -0,0 +1,200 @@
+# Custom Authentication
+
+This proposal focuses on supporting custom authentication for SASL/mTLS in the Strimzi operator. 
```

Comment:
Maybe to make things a bit more clear later ... could you make it clear in the title and maybe in the first paragraph that it is the broker / server part? Maybe _Custom Authentication in Kafka brokers_ would be a more fitting name.